src/lib.rs

```diff
@@ -687,6 +687,16 @@ pub enum ShaderModuleSource<'a> {
     Wgsl(Cow<'a, str>),
 }
 
+/// Descriptor for a shader module.
+pub struct ShaderModuleDescriptor<'a> {
+    /// Debug label of the shader module. This will show up in graphics debuggers for easy identification.
+    pub label: Label<'a>,
+    /// Source code for the shader.
+    pub source: ShaderSource<'a>,
+    /// Experimental translation path attempts to avoid SPIR-V and work with Naga IR directly.
+    pub experimental_translation: bool,
```

Comment:
If we're mostly using this internally, then another option is to enable this through an environment variable. Then we wouldn't need to change the API later when we're ready to remove the SPIR-V path